src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -268,6 +270,7 @@ create_reference_table(PG_FUNCTION_ARGS)
 	 * backends manipulating this relation.
 	 */
 	Relation relation = relation_open(relationId, ExclusiveLock);
+	relation_close(relation, NoLock);
```

Comment:
could directly use LockRelation or something, or should comment why we close relation immediately

Reply:
now that we use `try_relation_open` and error out if relation does not exist, resolving my comment